src/onboarding/registration/NameAndPicture.test.tsx

```diff
@@ -245,6 +245,25 @@ describe('NameAndPictureScreen', () => {
     expect(AccountActions.setName).not.toHaveBeenCalled()
   })
 
+  it('shows generator button and generator button puts random username in', () => {
+    mockStatsigGet.mockReturnValueOnce(false).mockReturnValueOnce('autogenerator')
+    const { getByText, getByTestId } = render(
+      <Provider store={createMockStore()}>
+        <NameAndPicture {...mockScreenProps} />
+      </Provider>
+    )
+    mockRandom.mockReturnValue(0)
+    expect(getByText('generateUsername')).toBeTruthy()
+    const nameField = getByTestId('NameEntry')
+    fireEvent.press(getByText('generateUsername'))
+    const text = nameField.props.value
+    mockRandom.mockReturnValue(0.9)
+    expect(text).toBeTruthy()
+    fireEvent.press(getByText('generateUsername'))
+    // different name on second press
+    expect(nameField.props.value).not.toEqual(text)
```

Comment:
why not mock the return value of the name generator and make sure the mocked value from the second invocation gets used here?

Reply:
mocking the random function seems like it makes this an integration test, testing the mock name generator helper function and the usage of it at the same time. I encourage you to prefer unit tests over integration tests-- they're easier to maintain and debug, and give you finer grained control over what is tested

Reply:
I absolutely agree that a test that just mocks what the user does instead of poking inside the implementation is better, which is what I wrote in the beginning, but the generator function would not work;
 I checked with Anthony and the way the random function is mocked made that difficult.